perf/stability/alertmanager/webhook.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/prometheus/alertmanager/template"
+)
+
+func webhook(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	data := template.Data{}
+	if err := json.NewDecoder(r.Body).Decode(&data); err != nil {
+		log.Fatalf(err.Error())
+		return
+	}
+	log.Printf("Alerts: GroupLabels=%v, CommonLabels=%v", data.GroupLabels, data.CommonLabels)
+	//TODO(richardwxn): process alert
+	for _, alert := range data.Alerts {
+		log.Printf("Alert: status=%s,Labels=%v,Annotations=%v", alert.Status, alert.Labels, alert.Annotations)
+	}
+	fmt.Fprint(w, "Ok!")
+}
+
+func healthz(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprint(w, "Ok!")
+}
+
+func main() {
```

Comment:
How does this work? We run this deployment and then alertmanager sends events and we do something with them?

Do we have a long term plan to do anything with the alerts aside from logging?

Reply:
yes, correct. I don't have a long term plan yet, but my initial thought is that besides logging, when user try to run it locally, notification should be sent to their email/slack, and when we run this on a pipeline then we can publish it to a dashboard? Any suggestion?

Reply:
slack seems pretty high value and easy to setup. For alerts we probably don't need historical data, right? So the Prometheus built in dashboard (or grafana if it will display these) with slack notification to direct us to the alert seems sufficient?

This is assuming we have persisted metrics in general so we don't need persisted alerts, just need to get notified?

Reply:
yep agreed, do you know who is our slack admin? I sent a request to configure the incoming webhook app for a new channel to test with